expression/builtin_compare.go

```diff
@@ -1059,6 +1059,9 @@ func getBaseCmpType(lhs, rhs types.EvalType, lft, rft *types.FieldType) types.Ev
 	} else if ((lhs == types.ETInt || lft.Hybrid()) || lhs == types.ETDecimal) &&
 		((rhs == types.ETInt || rft.Hybrid()) || rhs == types.ETDecimal) {
 		return types.ETDecimal
+	} else if types.IsTemporalWithDate(lft.Tp) && rhs == types.ETInt && rft.Tp == mysql.TypeYear ||
```

Comment:
Can `rhs == types.ETInt` and `lhs == types.ETInt` be removed?

Reply:
yes, it's redundant.